apps/gitness/src/pages-v2/repo/repo-commits.tsx

```diff
@@ -24,7 +25,22 @@ export default function RepoCommitsPage() {
     queryParams: { page }
   })
 
-  const [selectedBranch, setSelectedBranch] = useState<BranchSelectorListItem>({ name: '', sha: '' })
+  const { branchList, setBranchList, selectedBranch, setSelectedBranch, setRepoId, setSpaceId } =
+    useBranchSelectorStore()
+
+  useEffect(() => {
+    branches &&
+      !branchList &&
+      setBranchList(
```

Comment:
we should not have this condition. it will prevent updating the branchlist once it has been set.
instead, the `setBranchList` function should take care of merging the old and new data while de-duplicating it.